rust/src/lib.rs

```diff
@@ -1828,12 +1901,29 @@ pub struct FfiWalletSummary {
 }
 
 impl FfiWalletSummary {
-    fn some(summary: WalletSummary<AccountId>) -> *mut Self {
+    fn some<P: Parameters>(
+        db_data: &WalletDb<rusqlite::Connection, P>,
+        summary: WalletSummary<AccountId>,
+    ) -> *mut Self {
         let (account_balances, account_balances_len) = {
             let account_balances: Box<[FfiAccountBalance]> = summary
                 .account_balances()
                 .iter()
-                .map(FfiAccountBalance::new)
+                .map(|(account_id, balance)| {
+                    let account_index = match db_data
+                        .get_account(*account_id)
+                        .expect("we got the summary from the wallet")
```

Comment:
This expectation should probably be an error; it indicates database corruption or failure of a query that is separate from the summary retrieval.

Reply:
Ah yep, this specific exception could be a transient error. The `Option::unwrap` can still be used.